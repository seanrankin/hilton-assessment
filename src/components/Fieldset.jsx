import styled from 'styled-components';

export const Fieldset = styled.fieldset`
  border: 2px solid ${props => (props.checked ? '#e7e7e7' : '#ccd0dc')};
  background-color: ${props => (props.checked ? 'white' : '#dbdbe3')};
  display: inline-block;
  max-width: 110px;
  font-size: 0.6em;
  border-radius: 8px;
  padding: 0;
`;
