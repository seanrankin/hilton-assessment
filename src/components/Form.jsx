import styled from 'styled-components';

export const Form = styled.div`
  padding: 25px 0 6px 6px;
`;
