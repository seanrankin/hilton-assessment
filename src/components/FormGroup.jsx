import styled from 'styled-components';

export const FormGroup = styled.div`
  display: inline-block;
  margin-right: 10px;
`;
