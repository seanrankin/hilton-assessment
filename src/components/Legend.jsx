import styled from 'styled-components';

export const Legend = styled.legend`
  float: left;
  width: 100%;
  font-weight: bold;
  border-radius: 6px 6px 0 0;
  padding: 0 0 0 3px;
  background-color: ${props => (props.checked ? '#e7e7e7' : '#dbdbe3')};
  span {
    display: inline-block;
    padding: ${props => (props.id === 1 ? '3px 0 3px 3px' : 0)};
  }
`;
