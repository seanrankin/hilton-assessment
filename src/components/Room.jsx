import React, { Component } from 'react';
import {
  setRoomEnabled,
  updateAdults,
  updateChildren,
} from '../actions/reservations';

import { Fieldset } from './Fieldset';
import { Form } from './Form';
import { FormGroup } from './FormGroup';
import { Label } from './Label';
import { Legend } from './Legend';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class Room extends Component {
  handleCheck = () => {
    const {
      id,
      room: { checked },
      rooms,
      setRoomEnabled,
      updateAdults,
      updateChildren,
    } = this.props;

    setRoomEnabled(id);

    // Reset dropdowns
    if (checked) {
      updateAdults(id, 1);
      updateChildren(id, 0);
    }

    // Handle toggle check rooms for before and after
    if (!checked && id > 2) {
      for (let i = id - 1; i > 1; i--) {
        if (!rooms[`room${i}`].checked) {
          setRoomEnabled(i);
        }
      }
    } else if (checked && id > 1 && id < 4) {
      for (let i = id + 1; i <= 4; i++) {
        if (rooms[`room${i}`].checked) {
          setRoomEnabled(i);
          updateAdults(i, 1);
          updateChildren(i, 0);
        }
      }
    }
  };

  handleSelect = name => event => {
    const { id, updateAdults, updateChildren } = this.props;
    const {
      target: { value },
    } = event;

    if (name === 'adults') {
      updateAdults(id, +value);
    } else {
      updateChildren(id, +value);
    }
  };

  render() {
    const {
      id,
      room: { adults, children, checked },
    } = this.props;

    return (
      <Fieldset checked={checked}>
        <Legend id={id} checked={checked}>
          <input
            type="checkbox"
            name={`room${id}enabled`}
            checked={checked}
            onChange={this.handleCheck}
            style={{ display: id === 1 ? 'none' : 'inline' }}
          />
          <span>{`Room ${id}`}</span>
        </Legend>
        <Form>
          <FormGroup>
            <Label htmlFor={`adults${id}`}>
              Adults
              <br />
              (18+)
            </Label>
            <select
              id={`adults${id}`}
              value={adults}
              disabled={!checked}
              onChange={this.handleSelect('adults')}
            >
              <option value={1}>1</option>
              <option value={2}>2</option>
            </select>
          </FormGroup>
          <FormGroup>
            <Label htmlFor={`children${id}`}>
              Children
              <br />
              (0-17)
            </Label>
            <select
              id={`children${id}`}
              value={children}
              disabled={!checked}
              onChange={this.handleSelect('children')}
            >
              <option value={0}>0</option>
              <option value={1}>1</option>
              <option value={2}>2</option>
            </select>
          </FormGroup>
        </Form>
      </Fieldset>
    );
  }
}

Room.propTypes = {
  id: PropTypes.number.isRequired,
  room: PropTypes.object.isRequired,
  rooms: PropTypes.object.isRequired,
  setRoomEnabled: PropTypes.func.isRequired,
  updateAdults: PropTypes.func.isRequired,
  updateChildren: PropTypes.func.isRequired,
};

const mapStateToProps = ({ reservations }, { id }) => {
  const room = reservations[`room${id}`];
  const rooms = reservations;

  return {
    rooms,
    room,
  };
};

export default connect(
  mapStateToProps,
  { setRoomEnabled, updateAdults, updateChildren }
)(Room);
