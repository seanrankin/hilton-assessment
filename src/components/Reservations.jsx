import React, { Component } from 'react';

import PropTypes from 'prop-types';
import Room from './Room';
import { connect } from 'react-redux';
import { setStateFromLocalStorage } from '../actions/reservations';

export class Reservations extends Component {
  componentDidMount() {
    const { setStateFromLocalStorage } = this.props;
    const reservations = localStorage.getItem('reservations');

    reservations && setStateFromLocalStorage(JSON.parse(reservations));
  }

  handleSubmit = e => {
    const { reservations } = this.props;

    e.preventDefault();
    localStorage.setItem('reservations', JSON.stringify(reservations));
  };

  render() {
    return (
      <form onSubmit={e => this.handleSubmit(e)}>
        <Room id={1} />
        <Room id={2} />
        <Room id={3} />
        <Room id={4} />
        <br />
        <input type="submit" />
      </form>
    );
  }
}

Reservations.propTypes = {
  reservations: PropTypes.object.isRequired,
  setStateFromLocalStorage: PropTypes.func.isRequired,
};

const mapStateToProps = ({ reservations }) => {
  return {
    reservations,
  };
};

export default connect(
  mapStateToProps,
  { setStateFromLocalStorage }
)(Reservations);
