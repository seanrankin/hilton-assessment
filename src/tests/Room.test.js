import { Form } from '../components/Form';
import { Legend } from '../components/Legend';
import React from 'react';
import ReactDOM from 'react-dom';
import { Room } from '../components/Room';
import { shallow } from 'enzyme';

describe('Room Component', () => {
  let wrapper;

  const props = {
    id: 2,
    room: { adults: 1, children: 0, checked: true },
    rooms: {},
    setRoomEnabled: () => {},
    updateAdults: () => {},
    updateChildren: () => {},
  };

  beforeEach(() => {
    wrapper = shallow(<Room {...props} />);
  });

  afterEach(() => {
    wrapper.setProps(props);
  });

  it('should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display the legend', () => {
    const legend = wrapper.find(Legend);
    expect(legend).toHaveLength(1);

    const [checkbox, label] = legend.props().children;
    expect(checkbox.type).toBe('input');
    expect(label.props.children).toBe('Room 2');
  });

  it('should display the selection inputs', () => {
    const form = wrapper.find(Form);
    expect(form).toHaveLength(1);

    const [adults, children] = form.props().children;
    const [adultLabel, adultSelect] = adults.props.children;
    const [childrenLabel, childrenSelect] = children.props.children;

    expect(adultLabel.props.children).toEqual(['Adults', <br />, '(18+)']);
    expect(adultSelect.type).toBe('select');
    expect(adultSelect.props.disabled).toBe(false);
    expect(adultSelect.props.children).toHaveLength(2);

    expect(childrenLabel.props.children).toEqual([
      'Children',
      <br />,
      '(0-17)',
    ]);
    expect(childrenSelect.type).toBe('select');
    expect(childrenSelect.props.disabled).toBe(false);
    expect(childrenSelect.props.children).toHaveLength(3);
  });

  it('should display readonly when unchecked', () => {
    wrapper.setProps({ room: { adults: 1, children: 0, checked: false } });

    const form = wrapper.find(Form);
    const [adults, children] = form.props().children;
    const [, adultSelect] = adults.props.children;
    const [, childrenSelect] = children.props.children;

    expect(adultSelect.props.disabled).toBe(true);
    expect(childrenSelect.props.disabled).toBe(true);
  });
});
