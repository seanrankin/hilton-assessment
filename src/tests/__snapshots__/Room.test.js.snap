// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Room
    id={2}
    room={
      Object {
        "adults": 1,
        "checked": true,
        "children": 0,
      }
    }
    rooms={Object {}}
    setRoomEnabled={[Function]}
    updateAdults={[Function]}
    updateChildren={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "checked": true,
      "children": Array [
        <ForwardRef
          checked={true}
          id={2}
        >
          <input
            checked={true}
            name="room2enabled"
            onChange={[Function]}
            style={
              Object {
                "display": "inline",
              }
            }
            type="checkbox"
          />
          <span>
            Room 2
          </span>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef>
            <ForwardRef
              htmlFor="adults2"
            >
              Adults
              <br />
              (18+)
            </ForwardRef>
            <select
              disabled={false}
              id="adults2"
              onChange={[Function]}
              value={1}
            >
              <option
                value={1}
              >
                1
              </option>
              <option
                value={2}
              >
                2
              </option>
            </select>
          </ForwardRef>
          <ForwardRef>
            <ForwardRef
              htmlFor="children2"
            >
              Children
              <br />
              (0-17)
            </ForwardRef>
            <select
              disabled={false}
              id="children2"
              onChange={[Function]}
              value={0}
            >
              <option
                value={0}
              >
                0
              </option>
              <option
                value={1}
              >
                1
              </option>
              <option
                value={2}
              >
                2
              </option>
            </select>
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": true,
          "children": Array [
            <input
              checked={true}
              name="room2enabled"
              onChange={[Function]}
              style={
                Object {
                  "display": "inline",
                }
              }
              type="checkbox"
            />,
            <span>
              Room 2
            </span>,
          ],
          "id": 2,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "name": "room2enabled",
              "onChange": [Function],
              "style": Object {
                "display": "inline",
              },
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Room 2",
            },
            "ref": null,
            "rendered": "Room 2",
            "type": "span",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  float: left;
  width: 100%;
  font-weight: bold;
  border-radius: 6px 6px 0 0;
  padding: 0 0 0 3px;
  background-color: ",
              [Function],
              ";
  span {
    display: inline-block;
    padding: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.legend",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "legend",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <ForwardRef
                htmlFor="adults2"
              >
                Adults
                <br />
                (18+)
              </ForwardRef>
              <select
                disabled={false}
                id="adults2"
                onChange={[Function]}
                value={1}
              >
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
              </select>
            </ForwardRef>,
            <ForwardRef>
              <ForwardRef
                htmlFor="children2"
              >
                Children
                <br />
                (0-17)
              </ForwardRef>
              <select
                disabled={false}
                id="children2"
                onChange={[Function]}
                value={0}
              >
                <option
                  value={0}
                >
                  0
                </option>
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
              </select>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  htmlFor="adults2"
                >
                  Adults
                  <br />
                  (18+)
                </ForwardRef>,
                <select
                  disabled={false}
                  id="adults2"
                  onChange={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                </select>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Adults",
                    <br />,
                    "(18+)",
                  ],
                  "htmlFor": "adults2",
                },
                "ref": null,
                "rendered": Array [
                  "Adults",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "(18+)",
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-ifAKCX",
                    "isStatic": true,
                    "rules": Array [
                      "
  display: block;
  margin-bottom: 4px;
",
                    ],
                  },
                  "displayName": "styled.label",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-ifAKCX",
                  "target": "label",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value={1}
                    >
                      1
                    </option>,
                    <option
                      value={2}
                    >
                      2
                    </option>,
                  ],
                  "disabled": false,
                  "id": "adults2",
                  "onChange": [Function],
                  "value": 1,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1",
                      "value": 1,
                    },
                    "ref": null,
                    "rendered": "1",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2",
                      "value": 2,
                    },
                    "ref": null,
                    "rendered": "2",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
  display: inline-block;
  margin-right: 10px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  htmlFor="children2"
                >
                  Children
                  <br />
                  (0-17)
                </ForwardRef>,
                <select
                  disabled={false}
                  id="children2"
                  onChange={[Function]}
                  value={0}
                >
                  <option
                    value={0}
                  >
                    0
                  </option>
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                </select>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Children",
                    <br />,
                    "(0-17)",
                  ],
                  "htmlFor": "children2",
                },
                "ref": null,
                "rendered": Array [
                  "Children",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "(0-17)",
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-ifAKCX",
                    "isStatic": true,
                    "rules": Array [
                      "
  display: block;
  margin-bottom: 4px;
",
                    ],
                  },
                  "displayName": "styled.label",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-ifAKCX",
                  "target": "label",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value={0}
                    >
                      0
                    </option>,
                    <option
                      value={1}
                    >
                      1
                    </option>,
                    <option
                      value={2}
                    >
                      2
                    </option>,
                  ],
                  "disabled": false,
                  "id": "children2",
                  "onChange": [Function],
                  "value": 0,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "0",
                      "value": 0,
                    },
                    "ref": null,
                    "rendered": "0",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1",
                      "value": 1,
                    },
                    "ref": null,
                    "rendered": "1",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2",
                      "value": 2,
                    },
                    "ref": null,
                    "rendered": "2",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
  display: inline-block;
  margin-right: 10px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  padding: 25px 0 6px 6px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  border: 2px solid ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  display: inline-block;
  max-width: 110px;
  font-size: 0.6em;
  border-radius: 8px;
  padding: 0;
",
        ],
      },
      "displayName": "styled.fieldset",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "fieldset",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "checked": true,
        "children": Array [
          <ForwardRef
            checked={true}
            id={2}
          >
            <input
              checked={true}
              name="room2enabled"
              onChange={[Function]}
              style={
                Object {
                  "display": "inline",
                }
              }
              type="checkbox"
            />
            <span>
              Room 2
            </span>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef>
              <ForwardRef
                htmlFor="adults2"
              >
                Adults
                <br />
                (18+)
              </ForwardRef>
              <select
                disabled={false}
                id="adults2"
                onChange={[Function]}
                value={1}
              >
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
              </select>
            </ForwardRef>
            <ForwardRef>
              <ForwardRef
                htmlFor="children2"
              >
                Children
                <br />
                (0-17)
              </ForwardRef>
              <select
                disabled={false}
                id="children2"
                onChange={[Function]}
                value={0}
              >
                <option
                  value={0}
                >
                  0
                </option>
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
              </select>
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": true,
            "children": Array [
              <input
                checked={true}
                name="room2enabled"
                onChange={[Function]}
                style={
                  Object {
                    "display": "inline",
                  }
                }
                type="checkbox"
              />,
              <span>
                Room 2
              </span>,
            ],
            "id": 2,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "name": "room2enabled",
                "onChange": [Function],
                "style": Object {
                  "display": "inline",
                },
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Room 2",
              },
              "ref": null,
              "rendered": "Room 2",
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  float: left;
  width: 100%;
  font-weight: bold;
  border-radius: 6px 6px 0 0;
  padding: 0 0 0 3px;
  background-color: ",
                [Function],
                ";
  span {
    display: inline-block;
    padding: ",
                [Function],
                ";
  }
",
              ],
            },
            "displayName": "styled.legend",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "legend",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <ForwardRef
                  htmlFor="adults2"
                >
                  Adults
                  <br />
                  (18+)
                </ForwardRef>
                <select
                  disabled={false}
                  id="adults2"
                  onChange={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                </select>
              </ForwardRef>,
              <ForwardRef>
                <ForwardRef
                  htmlFor="children2"
                >
                  Children
                  <br />
                  (0-17)
                </ForwardRef>
                <select
                  disabled={false}
                  id="children2"
                  onChange={[Function]}
                  value={0}
                >
                  <option
                    value={0}
                  >
                    0
                  </option>
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                </select>
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef
                    htmlFor="adults2"
                  >
                    Adults
                    <br />
                    (18+)
                  </ForwardRef>,
                  <select
                    disabled={false}
                    id="adults2"
                    onChange={[Function]}
                    value={1}
                  >
                    <option
                      value={1}
                    >
                      1
                    </option>
                    <option
                      value={2}
                    >
                      2
                    </option>
                  </select>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Adults",
                      <br />,
                      "(18+)",
                    ],
                    "htmlFor": "adults2",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Adults",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "(18+)",
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-ifAKCX",
                      "isStatic": true,
                      "rules": Array [
                        "
  display: block;
  margin-bottom: 4px;
",
                      ],
                    },
                    "displayName": "styled.label",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-ifAKCX",
                    "target": "label",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value={1}
                      >
                        1
                      </option>,
                      <option
                        value={2}
                      >
                        2
                      </option>,
                    ],
                    "disabled": false,
                    "id": "adults2",
                    "onChange": [Function],
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "1",
                        "value": 1,
                      },
                      "ref": null,
                      "rendered": "1",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "2",
                        "value": 2,
                      },
                      "ref": null,
                      "rendered": "2",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
  display: inline-block;
  margin-right: 10px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef
                    htmlFor="children2"
                  >
                    Children
                    <br />
                    (0-17)
                  </ForwardRef>,
                  <select
                    disabled={false}
                    id="children2"
                    onChange={[Function]}
                    value={0}
                  >
                    <option
                      value={0}
                    >
                      0
                    </option>
                    <option
                      value={1}
                    >
                      1
                    </option>
                    <option
                      value={2}
                    >
                      2
                    </option>
                  </select>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Children",
                      <br />,
                      "(0-17)",
                    ],
                    "htmlFor": "children2",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Children",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "(0-17)",
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-ifAKCX",
                      "isStatic": true,
                      "rules": Array [
                        "
  display: block;
  margin-bottom: 4px;
",
                      ],
                    },
                    "displayName": "styled.label",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-ifAKCX",
                    "target": "label",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value={0}
                      >
                        0
                      </option>,
                      <option
                        value={1}
                      >
                        1
                      </option>,
                      <option
                        value={2}
                      >
                        2
                      </option>,
                    ],
                    "disabled": false,
                    "id": "children2",
                    "onChange": [Function],
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "0",
                        "value": 0,
                      },
                      "ref": null,
                      "rendered": "0",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "1",
                        "value": 1,
                      },
                      "ref": null,
                      "rendered": "1",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "2",
                        "value": 2,
                      },
                      "ref": null,
                      "rendered": "2",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
  display: inline-block;
  margin-right: 10px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  padding: 25px 0 6px 6px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            "
  border: 2px solid ",
            [Function],
            ";
  background-color: ",
            [Function],
            ";
  display: inline-block;
  max-width: 110px;
  font-size: 0.6em;
  border-radius: 8px;
  padding: 0;
",
          ],
        },
        "displayName": "styled.fieldset",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "fieldset",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
